pkg/util/release/max_surge.go

```diff
@@ -0,0 +1,33 @@
+package release
+
+import (
+	"math"
+
+	intstrutil "k8s.io/apimachinery/pkg/util/intstr"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+)
+
+func GetMaxSurgePercent(rollingUpdate *shipper.RollingUpdate, replicaCount int32) (int, error) {
+	maxSurgeDefaultValue := intstrutil.FromString("100%")
+	maxSurgeValue := intstrutil.FromString("100%")
+	if rollingUpdate != nil {
+		maxSurgeValue = *intstrutil.ValueOrDefault(&rollingUpdate.MaxSurge, maxSurgeValue)
+	}
+
+	surge, err := intstrutil.GetValueFromIntOrPercent(&maxSurgeValue, int(replicaCount), true)
+	if err != nil {
+		return 100, err
```

Comment:
Not that it matters a whole bunch since one is not supposed to use the returned value in case of error, but It's idiomatic to return a zero-value in that case.